$MirOS: ports/mailnews/pine/patches/patch-pine_mailcmd_c,v 1.2 2005/05/04 23:37:40 tg Exp $
--- pine/mailcmd.c.orig	Mon Sep 12 22:04:03 2005
+++ pine/mailcmd.c	Fri Oct  7 19:12:54 2005
@@ -1231,7 +1231,7 @@ get_out:
 	      break;
 	}
 
-	dprint(8,(debugfile, "Expunge max:%ld cur:%ld kill:%d\n",
+	dprint(8,(debugfile, "Expunge max:%ld cur:%ld kill:%ld\n",
 		  mn_get_total(msgmap), mn_get_cur(msgmap), del_count));
 
 	old_max_msgno = mn_get_total(msgmap);
@@ -5601,7 +5601,7 @@ cmd_export(state, msgmap, qline, agg)
 	    }
 
 	    ok = 0;
-	    sprintf(dir, "%s.d", full_filename, S_FILESEP);
+	    sprintf(dir, "%s.d", full_filename);
 
 	    do {
 		tries++;
@@ -5700,7 +5700,7 @@ cmd_export(state, msgmap, qline, agg)
 		if(strlen(dir) + strlen(S_FILESEP) + strlen(lfile) + 1
 							    > sizeof(filename)){
 		    dprint(2, (debugfile,
-			   "FAILED Att Export: name too long: %s\n",
+			   "FAILED Att Export: name too long: %s%s%s\n",
 			   dir, S_FILESEP, lfile));
 		    errs++;
 		    continue;
@@ -7367,6 +7367,7 @@ do_broach_folder(newfolder, new_context,
     SEARCHSET  *srchset = NULL;
     unsigned    perfolder_startup_rule;
     char        tmp1[MAILTMPLEN], tmp2[MAILTMPLEN], *lname, *mname;
+    char	*xtmp;
 
 #if	defined(DOS) && !defined(WIN32)
     openmode = OP_SHORTCACHE | SP_USERFLDR;
@@ -7665,10 +7666,10 @@ do_broach_folder(newfolder, new_context,
 	if(was_dead)
 	  icon_text(NULL, IT_MCLOSED);
 
-	dprint(7, (debugfile, "%ld %ld %x\n",
+	dprint(7, (debugfile, "%ld %ld %lx\n",
 		   mn_get_cur(ps_global->msgmap),
                    mn_get_total(ps_global->msgmap),
-		   ps_global->mail_stream));
+		   (unsigned long)ps_global->mail_stream));
 	/*
 	 * remember last context and folder
 	 */
@@ -7687,9 +7688,9 @@ do_broach_folder(newfolder, new_context,
 	    ps_global->last_unambig_folder[sizeof(ps_global->last_unambig_folder)-1] = '\0';
 	}
 
-	strncpy(ps_global->cur_folder, sp_fldr(ps_global->mail_stream)
-					 ? sp_fldr(ps_global->mail_stream)
-					 : ps_global->inbox_name,
+	xtmp = sp_fldr(ps_global->mail_stream) ? sp_fldr(ps_global->mail_stream)
+	    : ps_global->inbox_name;
+	strncpy(ps_global->cur_folder, xtmp,
 		sizeof(ps_global->cur_folder)-1);
 	ps_global->cur_folder[sizeof(ps_global->cur_folder)-1] = '\0';
 	ps_global->context_current = ps_global->context_list;
@@ -7844,10 +7845,10 @@ do_broach_folder(newfolder, new_context,
                            mn_get_cur(ps_global->msgmap),
                            mn_get_total(ps_global->msgmap),
                            ps_global->mail_stream));
-		strncpy(ps_global->cur_folder,
-			sp_fldr(ps_global->mail_stream)
+		xtmp =	sp_fldr(ps_global->mail_stream)
 			  ? sp_fldr(ps_global->mail_stream)
-			  : ps_global->inbox_name,
+			  : ps_global->inbox_name;
+		strncpy(ps_global->cur_folder, xtmp,
 			sizeof(ps_global->cur_folder)-1);
 		ps_global->cur_folder[sizeof(ps_global->cur_folder)-1] = '\0';
             }
@@ -12243,7 +12244,7 @@ select_date(stream, msgmap, msgno, limit
 	seldate_tm = localtime(&seldate);
 	sprintf(defdate, "%.2d-%.4s-%.4d", seldate_tm->tm_mday,
 		month_abbrev(seldate_tm->tm_mon + 1),
-		seldate_tm->tm_year + 1900);
+		(int)seldate_tm->tm_year + 1900);
 	sprintf(prompt,"Select messages which %s%s%s [%s]: ",
 		tense[when].preamble, tense[when].range,
 		tense[when].scope, defdate);
